ictv/templates/users.html

```diff
@@ -91,69 +99,81 @@ <h4><i class="icon fa $icon"></i>$action.title()</h4>
-                                        <h6 class="label bg-purple" $:make_tooltip(tooltip_message)><i class="fa fa-user"></i>Channel administrator</h6>
+                                    {% if admin_channels %}
+                                        <h6 class="label bg-purple" {{ make_tooltip(tooltip_message) | safe}}><i class="fa fa-user"></i>Channel administrator</h6>
+                                    {% endif %}
 
-                                    $code:
-                                        contrib_channels = list(user.get_channels_with_permission_level(UserPermissions.channel_contributor))
-                                        channel_names = '\n'.join([c.name for c in contrib_channels])
-                                        tooltip_message = '%s is contributor of the following channels:\n%s' % (user.fullname if user.fullname else user.email, channel_names)
+                                    {% set contrib_channels = (user.get_channels_with_permission_level(UserPermissions.channel_contributor) | list) %}
+                                    {% set channel_names = (contrib_channels | join('\n',attribute="name")) %}
+                                    {% set tooltip_message = '%s is contributor of the following channels:\n%s' % (user.fullname if user.fullname else user.email, channel_names) %}
 
-                                    $if contrib_channels:
-                                        <h6 class="label label-info" $:make_tooltip(tooltip_message)><i class="fa fa-user"></i>Channel contributor</h6>
+                                    {% if contrib_channels %}
+                                        <h6 class="label label-info" {{ make_tooltip(tooltip_message) | safe }}><i class="fa fa-user"></i>Channel contributor</h6>
+                                    {% endif %}
 
-                                    $code:
-                                        screens = list(user.screens)
-                                        screen_names = '\n'.join(['%s, (%s)' % (s.name, s.building.name) for s in screens])
-                                        tooltip_message = '%s is administrator of the following screens:\n%s' % (user.fullname if user.fullname else user.email, screen_names)
+                                    {% set screens = (user.screens | list) %}
+                                    {% set tooltip_message = ['%s is administrator of the following screens:\n' % (user.fullname if user.fullname else user.email)] %}
 
-                                    $if screens:
-                                        <h6 class="label bg-olive" $:make_tooltip(tooltip_message)><i class="fa fa-user"></i>Screen administrator</h6>
+                                    {# TODO seek a more elegant workaround #}
+                                    {% for s in screens %}
+                                      <div style="display:none">{{ tooltip_message.append("%s, (%s)\n" | format(s.name, s.building.name)) }}</div>
```

Comment:
Why this has to be added ?

Reply:
**xxx.append()** in jinja2 is an expression. Using it as a statement would require the activation of an extension.
(https://jinja.palletsprojects.com/en/3.0.x/extensions/#expression-statement)

Simply redirecting the output to a silenced div was a quick way to recreate the previous layout without writing "backend" code since it might change during the flask transition.

Reply:
Can't you simply use a comprehension and list concatenation to avoid using ``append`` ?

``{% set tooltip_message = ['first tooltip'] + ['another tooltip' for s in screens] %}`` or something similar ?

Reply:
Comprehension lists do not work in jinja2, I might work something out with the filters tho.